### PR TITLE
add check if server url containing http, if not add it

### PR DESCRIPTION
**Pull request by ronshuli**

### CODE REVIEW

**Comment by ronshuli** [comment]
Hi this is my first PR,
I fixed this bug https://github.com/prometheus/prometheus/issues/6397
thanks,

**Comment by roidelapluie** [comment]
Thank you for your pull request @ronshuli.

This does not seem to be the right fix however. I think we should do that earlier, at the point where we parse the parameter, so that it will benefit all the commands.

Ping me if you need more assitance to move forward, or when you have updated your pull request, and once again, welcome on the Prometheus project!

**Comment by ronshuli** [comment]
Hello @roidelapluie,
I think I fixed the PR according what you asked,
I would love that you will check this once again and thank for notice.

I'm happy to join Prometheus project :)

**Comment by roidelapluie** [comment]
I will review in a few days ; however I think the approach should be to parse the URL and see if there is a protocol ; instead of checking for a string. I will come back to you :)

Also, did you test this? I was under the impression that we check the URL when we parse the arguments, using prometheus/common

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -124,7 +125,12 @@ func main() {
 		os.Exit(CheckMetrics())
 
 	case queryInstantCmd.FullCommand():
-		os.Exit(QueryInstant(*queryServer, *queryExpr, p))
+		{
+			if !strings.Contains(*queryServer, "http") {
```
Comment (codesome) [inline cmd/promtool/main.go]: I am not aware of the placement of this check or any other ways, but if we are going with string matching, then I think only the prefix should be checked for `http://` and `https://` and not the entire URL for the existence of `http`.

Comment (ronshuli) [reply cmd/promtool/main.go]: Hi, thanks for your comment, I change it :)

**Comment by roidelapluie** [comment]
hello,

Somehow this has been addressed in the meantime: #7956

Thanks for your work and sorry this went un-noticed!

**Pull request closed without merging**